tests/test_fastavro.py

```diff
@@ -595,6 +596,119 @@ def test_schema_migration_reader_union():
     assert new_records == records
 
 
+def test_aliases_in_reader_schema():
+    schema = {
+        "type": "record",
+        "fields": [{
+            "name": "test",
+            "type": "int"
+        }]
+    }
+
+    new_schema = {
+        "type": "record",
+        "fields": [{
+            "name": "newtest",
+            "type": "int",
+            "aliases": ["test"]
+        }]
+    }
+
+    new_file = MemoryIO()
+    records = [{"test": 1}]
+    fastavro.writer(new_file, schema, records)
+    new_file.seek(0)
+    new_reader = fastavro.reader(new_file, new_schema)
+    new_records = list(new_reader)
+    assert new_records[0]["newtest"] == records[0]["test"]
+
+
+def test_unused_aliases_in_reader_schema():
+    schema = {
+        "type": "record",
+        "fields": [
+            {
+                "name": "test1",
+                "type": "int"
+            },
+            {
+                "name": "test2",
+                "type": "int"
+            }
+        ]
+    }
+
+    new_schema = {
+        "type": "record",
+        "fields": [{
+            "name": "test2",
+            "type": "int",
+            "aliases": ["test1"]
+        }]
+    }
+
+    new_file = MemoryIO()
+    records = [{"test1": 1, "test2": 2}]
+    fastavro.writer(new_file, schema, records)
+    new_file.seek(0)
+    new_reader = fastavro.reader(new_file, new_schema)
+    new_records = list(new_reader)
+    assert new_records[0]["test2"] == records[0]["test2"]
```

Comment:
I'm not sure about this test. The specification for aliases states `Aliases function by re-writing the writer's schema using aliases from the reader's schema. For example, if the writer's schema was named "Foo" and the reader's schema is named "Bar" and has an alias of "Foo", then the implementation would act as though "Foo" were named "Bar" when reading.`

That seems to imply to me that we should treat the original schema's `test1` field as being named `test2` (and we would ignore the original schema's `test2` field) so the assertion should be `assert new_records[0]["test2"] == records[0]["test1"]`

What are your thoughts?

Reply:
I must admit that I slightly interpreted the documentation: the alias is used only if a field with the original name is not present. This is confirmed by the implementation of the `avro` python package:

 ```
import avro.schema
from avro.datafile import DataFileReader, DataFileWriter
from avro.io import DatumReader, DatumWriter

# Write data
writers_schema = avro.schema.make_avsc_object({
    "type": "record",
    "name": "pouac",
    "fields": [
        {
            "name": "test1",
            "type": "int"
        },
        {
            "name": "test2",
            "type": "int"
        }
    ]
})
writer = DataFileWriter(open("records.avro", "wb"), DatumWriter(), writers_schema)
writer.append({"test1": 1, "test2": 2})
writer.close()

# Read data
readers_schema = avro.schema.make_avsc_object({
    "type": "record",
    "name": "pouac",
    "fields": [{
        "name": "test2",
         "type": "int",
         "aliases": ["test1"]
    }]
})
reader = DataFileReader(open("records.avro", "rb"), DatumReader(readers_schema=readers_schema))
for user in reader:
    print user
reader.close()
```

This script prints `{u'test2': 2}`, so the `avro` package has the same behaviour.

Reply:
The python version of the standard `avro` package doesn't implement aliasing (https://issues.apache.org/jira/browse/AVRO-1303). I think I read that the java code does implement it, so it would be interesting to see what happens on the java side.

Reply:
HA! I didn't know that. I'll summon old-forgotten Java skills and check the Java implementation.